src/test/ui/consts/const_unsafe_unreachable.rs

```diff
@@ -1,18 +1,28 @@
 // run-pass
 
+#![feature(core_intrinsics)]
 #![feature(const_fn)]
 #![feature(const_if_match)]
 #![feature(const_unreachable_unchecked)]
 
-const unsafe fn foo(x: u8) -> u8 {
+const unsafe fn f_std_hint_unreachable(x: u8) -> u8 {
     match x {
         42 => 34,
         _ => std::hint::unreachable_unchecked(),
     }
 }
 
-const BAR:u8 = unsafe { foo(42) };
+const unsafe fn f_std_intrinsics_unreachable(x: u8) -> u8 {
+    match x {
+        17 => 22,
+        _ => std::intrinsics::unreachable(),
```

Comment:
`std::hint::unreachable_unchecked` is a thing wrapper around the intrinsic, I do not see the point of also testing the intrinsic directly? I don't think we usually do that.

The test @eddyb and me asked for was a case where `unreachable_unchecked` is used *incorrectly*, where UB is triggered by actually reaching that function. Like, `f_std_hint_unreachable(23)`.